perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/UsersManagerMethod.java

```diff
@@ -1321,6 +1321,19 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 				}
 				referer = customUrl;
 				customPath = url.getPath();
+			} else { // referer + linkPath option
+				// check that path won't change domain of the url (security risk)
+				if (customPath != null) {
+					try {
+						URL refUrl = new URL(referer);
+						URL refDomainWithPath = new URL(refUrl.getProtocol()+"://"+refUrl.getHost()+customPath);
+						if (!refUrl.getHost().equals(refDomainWithPath.getHost())) {
+							throw new RpcException(RpcException.Type.INVALID_URL, "Invalid verification link - path changes domain: " + refDomainWithPath);
+						}
+					} catch (MalformedURLException e) {
+						throw new RpcException(RpcException.Type.INVALID_URL, "Invalid referer or path.");
+					}
+				}
```

Comment:
I really like this solution. However, we should also support the combination of `customUrl` and `customPath` being provided at the same time. I have discovered this problem today and I have created a PR with a fix: https://github.com/CESNET/perun/pull/3221

Could you please review it, so I can merge it? Then you probably should rebase this PR.

Also, I think you can move this check out of the first `if`. You can validate this regardless of the `customUrl` being provided or not.

Reply:
After agreeing to further keep only customUrl and no linkPath, I only edited the else - if part.